docs/devel/controllers.md

```diff
@@ -0,0 +1,101 @@
+# Writing Controllers
+
+A Kubernetes controller is simply an optimization on bash script that does repeated lists against the API server.
```

Comment:
> Why a bash script? :)

Easy example that nearly everyone is familiar with and no one actually uses any other shell, right? :)